pyramid_oereb/lib/readers/extract.py

```diff
@@ -115,7 +115,18 @@ def read(self, params, real_estate, municipality):
                     for ds in plr_source.datasource:
                         if not params.skip_topic(ds.theme.code):
                             datasource.append(ds)
-                    real_estate.public_law_restrictions.extend(plr_source.records)
+
+                    # Group PLR records according to their law status
+                    grouped_plr = list()
+                    for law_status in Config.get_law_statuses():
+                        for plr in plr_source.records:
+                            if isinstance(plr, PlrRecord):
+                                if plr.law_status.code == law_status:
+                                    grouped_plr.append(plr)
+                            elif plr not in grouped_plr:
+                                grouped_plr.append(plr)
```

Comment:
This if does not make too much sense to me:
 you are doing the same thing whether it is `isinstance(plr, PlrRecord) and plr.law_status.code == law_status` OR it is a `plr not in grouped_plr`
Can't you write something like this:
```
if (isinstance(plr, PlrRecord) and plr.law_status.code == law_status) or plr not in grouped_plr:
    grouped_plr.append(plr)
```
I did not check the details so I might have missed something but it looks odd to have an if-elif that does the same thing.

Reply:
Thanks for pointing this out. I tried 
> if (isinstance(plr, PlrRecord) and plr.law_status.code == law_status) or plr not in grouped_plr:
>     grouped_plr.append(plr)

but this would append some `PlrRecord` with the wrong law status code in case they are not already in `grouped_plr`. The `elif` is for `EmptyPlrRecord` (which don't have a law status but we still need to add them to `grouped_plr`, just once). I tried to find a way to reformulate the conditions but did not succeed - suggestions are welcome. Maybe the sorting approach could be improved